packages/flutter_tools/test/general.shard/build_system/targets/asset_transformer_test.dart

```diff
@@ -0,0 +1,377 @@
+      dartBinaryPath: dartBinaryPath,
+    );
+
+    await transformer.transformAsset(
+      asset: asset,
+      outputPath: outputPath,
+      workingDirectory: fileSystem.currentDirectory.path,
+      transformerEntries: <AssetTransformerEntry>[
+        const AssetTransformerEntry(
+          package: 'my_transformer',
+          args: <String>[],
+        )
+      ],
+    );
+
+    expect(processManager, hasNoRemainingExpectations);
+    expect(logger.errorText, contains(RegExp(
+'''
+User-defined transformation of asset "asset\\.txt" failed\\.
+Transformer process terminated with non-zero exit code: 1
+Transformer package: my_transformer
+Full command: $dartBinaryPath run my_transformer --input=.*\\.txt --output=.*\\.txt
+stdout:
+Beginning transformation
+stderr:
+Something went wrong
+''')));
+  });
+
+  testWithoutContext('prints error message when the transformer does not produce an output file', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
+    final BufferLogger logger = BufferLogger.test();
+    final Artifacts artifacts = Artifacts.test();
+
+    final File asset = fileSystem.file('asset.txt')..createSync();
+    const String outputPath = 'output.txt';
+
+    final String dartBinaryPath = artifacts.getArtifactPath(Artifact.engineDartBinary);
+    final FakeProcessManager processManager =
+        FakeProcessManager.list(<FakeCommand>[
+      FakeCommand(
+        command: <Pattern>[
+          dartBinaryPath,
+          'run',
+          'my_transformer',
+          RegExp('--input=.*'),
+          RegExp('--output=.*'),
+        ],
+        onRun: (_) {
+          // Do nothing.
+        },
+        stderr: 'Transformation failed, but I forgot to exit with a non-zero code.'
+      ),
+    ]);
+
+    final AssetTransformer transformer = AssetTransformer(
+      processManager: processManager,
+      logger: logger,
+      fileSystem: fileSystem,
+      dartBinaryPath: dartBinaryPath,
+    );
+
+    await transformer.transformAsset(
+      asset: asset,
+      outputPath: outputPath,
+      workingDirectory: fileSystem.currentDirectory.path,
+      transformerEntries: <AssetTransformerEntry>[
+        const AssetTransformerEntry(
+          package: 'my_transformer',
+          args: <String>[],
+        )
+      ],
+    );
+
+    expect(processManager, hasNoRemainingExpectations);
+    expect(logger.errorText, contains(RegExp(
+r'''
+User-defined transformation of asset "asset\.txt" failed.
+Asset transformer my_transformer did not produce an output file\.
+Input file provided to transformer: ".*\.txt"
+Expected output file at: ".*\.txt"
+Full command: Artifact\.engineDartBinary run my_transformer --input=.*\.txt --output=.*\.txt
+stdout:
+
+stderr:
+Transformation failed, but I forgot to exit with a non-zero code\.
+'''
+    )));
+  });
+
+  testWithoutContext('correctly chains transformations when there are multiple of them', () async {
+    final FileSystem fileSystem = MemoryFileSystem();
```

Comment:
`MemoryFileSystem.test()`

Reply:
Addressed in [nit pack 1](https://github.com/flutter/flutter/pull/143815/commits/898a4b6396aee4098bce759b35caec356c96809a)